### ISSUE TITLE
Interpolate table values without if-else statements

### ISSUE THREAD

**Issue by paulobruno** [natema/WorldDynamics.jl#23]
The [recommended way](https://github.com/SciML/ModelingToolkit.jl/blob/master/docs/src/basics/FAQ.md#how-do-i-handle-if-statements-in-my-symbolic-forms) of using an if-else check in ModellingToolkit is to use the [`IfElse.ifelse`](https://github.com/SciML/IfElse.jl) function.
However, this is not possible in our interpolate function, because we can potentially evaluate a position outside of the range. If this is the case, we get a BoundsError.

Currently, our workaround is to use Tuples, which allows the interpolate function to be called only after the variables were already evaluated.

**Comment by paulobruno** [comment]
In b45cd29, I created a branchless version of interpolate using a boolean mathematical expression. This solves the problem, but the solution now is very very slow.

**Comment by paulobruno** [comment]
Using explicit types in the moment of registration, as done in ab2c0e8, gives us a huge speed-up.

**Comment by natema** [comment]
I wonder if, instead of using a loop as in b45cd29, construting the linear interpolation by building the expression with `Meta.parse` and then feeding it into `Meta.eval` could improve.

**Comment by paulobruno** [comment]
@natema Since the problem was effectively solved, I'm closing this issue. It would be great if you can create a new issue for the suggestion of using metaprogramming.
